### ISSUE TITLE
Command Line Arguments and Import issue

### ISSUE THREAD

**Issue by djnugent** [dronekit/dronekit-python#13]
Is it possible to have command line arguments for calling a script using api start?
Example "api start SmartCamera.py land -c 0"
When run I get "usage: api load <filename>"

Also when I run my script(without arguments), my script has problems completing imports. However, my script is able to complete imports when I run it normally(not with droneapi). Note: even if the script did complete imports it would crash due to lack of arguments. 

I don't think I'm fully understanding the usage.

If it is helpful my program can be seen at github.com/djnugent/SmartCamera

Thanks,
Daniel

**Comment by tcr3dr** [comment]
@djnugent Is this still an issue? Should this be documented better?

**Comment by hamishwillee** [comment]
@tcr3dr @djnugent This is still an issue because I haven't documented command line arguments to api start (at all) and I don't understand enough from this post to do so :-(

1. When you say this takes its  arguments from `"mavproxy.py [args]" do you mean that when you run a script with ``api start filename.py`` the arguments recieved will be whatever you already passed when you started mavproxy? In most cases that will just be the id of the master/connection?

  * If so, can you pass arbitrary commands to mavproxy and still pick them up in your script (ie launch mavproxy with parameters that mean nothing to mavproxy but which are relevant to the script?

2. The full question here is "how to interact with the script" - both with command line arguments and by making the script interactive - ie able to wait on and take some form of input. I might (for example) want to be able to send a message to my script from a GCS telling it to change behaviour in some way. Is this reasonable, and if so, can we have a separate issue for it?

**Comment by tcr3dr** [comment]
So I believe the issue is more specifically, "What is sys.argv populated with?" and the answer for this is "The arguments passed to mavproxy.py".

To pass "arguments" to your script in the current configuration, the best choice would be environment variables. We could extend `api start` to fix this.

However, I think punting on this issue is preferable, as in a post-MAVProxy  scenario we'd be able to control `sys.argv` and `stdin`, leaving it implementation-specific. As such, I'm marking this wontfix.

**Comment by hamishwillee** [comment]
@tcr3dr Makes sense to me. #188 will make a lot of things more "sensible".

**Comment by hamishwillee** [comment]
@tcr3dr Reopening to continue discussion.

Just found that in theory at least you can pass arguments to the script! See http://python.dronekit.io/examples/flight_replay.html
```
mission_id = int(local_arguments[0])
```

Any ideas of best place to document this in our current structure? We don't (yet) have best practices or programming idioms doc. Maybe a stand alone doc "Communicating with the script". WOuld be a bit "thin".

**Comment by tcr3dr** [comment]
We can now close this, since a public build of DKPY is in the wild.

Recommendation: use `sys.argv`!
